Rules/CompatibilityRules/CompatibilityRule.cs

```diff
@@ -0,0 +1,198 @@
+        /// but a non-constant expression cannot be used as an attribute parameter.
+        /// This is done in the ConfigureRule() override below.
+        /// The ConfigurableRuleProperty should just remove this parameter and use the
+        /// property default value.
+        /// </remarks>
+        [ConfigurableRuleProperty(defaultValue: "")]
+        public string AnyProfilePath { get; set; }
+
+        [ConfigurableRuleProperty(defaultValue: new string[] {})]
+        public string[] TargetProfilePaths { get; set; }
+
+        public virtual DiagnosticSeverity DiagnosticSeverity => DiagnosticSeverity.Warning;
+
+        protected abstract CompatibilityVisitor CreateVisitor(string fileName);
+
+        public override IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
+        {
+            CompatibilityVisitor compatibilityVisitor = CreateVisitor(fileName);
+            ast.Visit(compatibilityVisitor);
+            return compatibilityVisitor.GetDiagnosticRecords();
+        }
+
+        public override RuleSeverity GetSeverity()
+        {
+            return RuleSeverity.Warning;
+        }
+
+        public override SourceType GetSourceType()
+        {
+            return SourceType.Builtin;
+        }
+
+        public override string GetSourceName()
+        {
+            return string.Format(CultureInfo.CurrentCulture, Strings.SourceName);
+        }
+
+        public override void ConfigureRule(IDictionary<string, object> paramValueMap)
+        {
+            base.ConfigureRule(paramValueMap);
+
+            // Default anyprofile path is the one specified in the CrossCompatibility library
+            if (string.IsNullOrEmpty(AnyProfilePath))
+            {
+                AnyProfilePath = PlatformNaming.AnyPlatformUnionName;
+            }
+        }
+
+        protected Tuple<CompatibilityProfileData, CompatibilityProfileData[]> LoadCompatibilityProfiles()
+        {
+            if (string.IsNullOrEmpty(AnyProfilePath))
+            {
+                throw new InvalidOperationException($"{nameof(AnyProfilePath)} cannot be null or empty");
+            }
+
+            if (TargetProfilePaths == null)
+            {
+                throw new InvalidOperationException($"{nameof(TargetProfilePaths)} cannot be null");
+            }
+
+            if (TargetProfilePaths.Length == 0)
+            {
+                throw new InvalidOperationException($"{nameof(TargetProfilePaths)} cannot be empty");
+            }
+
+            var targetProfiles = new List<CompatibilityProfileData>();
+            foreach (string configPath in TargetProfilePaths)
+            {
+                string normalizedPath = NormalizeProfileNameToAbsolutePath(configPath);
+                targetProfiles.Add(_profileLoader.GetProfileFromFilePath(normalizedPath));
+            }
+
+            CompatibilityProfileData anyProfile = _profileLoader.GetProfileFromFilePath(NormalizeProfileNameToAbsolutePath(AnyProfilePath));
```

Comment:
One possibility is to not include the anyprofile at all and instead generate it on first launch. This would be expensive and to be performant would need to write to the file system, but would mean adding a new profile always "just works" and saves us shipping a 70MB file.

Reply:
Generating the union could take a while, but I've made the changes so that once it's generated, it can verify that it doesn't need regenerating.

Agreed that I don't like shipping the 70MB file too.

I'll have a look into auto-creation of this.

Reply:
Sounds good. Would it make sense to do it only if the rule actually gets run since a lot of users will never configure this rule to be run?

Reply:
Sounds good. Would it make sense to do it only if the rule actually gets run since a lot of users will never configure this rule to be run?